Lib/ufo2ft/makeotfParts.py

```diff
@@ -36,20 +40,9 @@ def compile(self):
         features. If they already exist, they will not be overwritten.
         """
 
-        self.precompile()
         self.setupFile_features()
         self.setupFile_featureTables()
-
-        # only after compiling features can usMaxContext be calculated
-        self.outline['OS/2'].usMaxContext = maxCtxFont(self.outline)
-
-    def precompile(self):
-        """Set any attributes needed before compilation.
-
-        **This should not be called externally.** Subclasses
-        may override this method if desired.
-        """
-        pass
```

Comment:
I'm thinking clients can override compile() if needed, add steps at the beginning.